pkg/download/download.go

```diff
@@ -57,14 +77,16 @@ func Decompressed(client *http.Client, u url.URL, path string) error {
 	default:
 		return fmt.Errorf("unknown extension: %s", u.Path)
 	}
+
 	if _, err := io.Copy(tmpFile, decompressor); err != nil {
 		return err
 	}
-	if err := tmpFile.Close(); err != nil {
-		return err
-	}
+
```

Comment:
I think this probably still needs a `Close` here before the move?

Reply:
Suprisingly, no you don't. You can rename before closing. https://go.dev/play/p/gX4dPvGtcQt

Reply:
In order to close the file before renaming, this could would become more complex. I would no longer be able to defer `tmpFile.Close()`. Closing twice is an error.

Reply:
Is it possible that there are un-synced writes if the file is not yet closed? Or does it not matter because the file descriptor doesn't change when the underlying filesystem link changes?

Also... Windows?

Reply:
Yes, it's possible that writes haven't been persisted to disk until you call close. Perhaps it would be better to move the `tmpFile.Close()` out of the defer so that we can handle the error.

> Or does it not matter because the file descriptor doesn't change when the underlying filesystem link changes?

Correct, the file descriptor doesn't change.

Reply:
Updated the code to handle the error from close in 2d099f3. I think closing the file twice is a benign issue, and handling the error explicitly from close before renaming is definitely more correct.